contextual_lenses/train_utils.py

```diff
@@ -130,8 +136,48 @@ def apply(self, x, encoder_fn, encoder_fn_kwargs, reduce_fn, reduce_fn_kwargs,
     return out
 
 
+def load_params(params, encoder_fn_params=None, reduce_fn_params=None, predict_fn_params=None):
+  """Updates randomly initialized parameters using loaded parameters."""
+
+  loaded_params = copy.deepcopy(params) 
+  fn_names = list(loaded_params.keys())
+  
+  num_learnable_layers = 1
+  if encoder_fn_params is not None:
+    num_learnable_layers += 1
+    encoder_fn_ind = '_0'
+    reduce_fn_ind = '_1'
+  else:
+    reduce_fn_ind = '_0'
+  if reduce_fn_params is not None:
+    num_learnable_layers += 1
+
+  assert(len(loaded_params.keys()) == num_learnable_layers), 'Model encoder and lens architecture incorrectly specified!'
+ 
+  if predict_fn_params is not None:
+    for fn_name in fn_names:
+      if '_' + str(len(fn_names)-1) in fn_name:
+        predict_fn_name = fn_name
+    loaded_params[predict_fn_name] = predict_fn_params
+    
+  if encoder_fn_params is not None:
+    for fn_name in fn_names:
+        if encoder_fn_ind in fn_name:
+            encoder_fn_name = fn_name
```

Comment:
Suggest checking in each of these for loops that encoder_fn_name (or predict_fn_name in previous one) is only set once.

Something like:
```
encoder_fn_name = None
for ...
  if encoder_fn_in in fn_name:
    if encoder_fn_name is not None:
      raise ValueError('Multiple instances of encoder detected. %s' % fn_name)
```

Good to have additional error checking to make sure it works as expected.